server/src/controllers/seasonController.ts

```diff
@@ -145,6 +145,78 @@ seasonController.get('/', async (req: Request, res: Response) => {
   }
 });
 
+/**
+ * Retrieves a list of seasons with their associated events.
+ *
+ * @swagger
+ * /2/season/list:
+ *   get:
+ *     summary: Get all seasons with events
+ *     tags:
+ *     - New season
+ *     responses:
+ *       200:
+ *         description: Successful response with an array of seasons and their events.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 type: object
+ *                 properties:
+ *                   seasonId:
+ *                     type: number
+ *                     description: The unique identifier of the season.
+ *                   seasonName:
+ *                     type: string
+ *                     description: The name of the season.
+ *                   events:
+ *                     type: array
+ *                     description: An array of events associated with the season.
+ *                     items:
+ *                       type: object
+ *                       properties:
+ *                         eventId:
+ *                           type: number
+ *                           description: The unique identifier of the event.
+ *                         eventName:
+ *                           type: string
+ *                           description: The name of the event.
```

Comment:
You'll probably have to update this now that you've changed what the API returns, but this block is very handy to have when trying to look for APIs to use

Reply:
updated